src/node_realm.cc

```diff
@@ -0,0 +1,300 @@
+#include "node_realm.h"
+#include "env-inl.h"
+
+#include "memory_tracker-inl.h"
+#include "node_builtins.h"
+#include "node_process.h"
+#include "util.h"
+
+namespace node {
+
+using builtins::BuiltinLoader;
+using v8::Context;
+using v8::EscapableHandleScope;
+using v8::Function;
+using v8::HandleScope;
+using v8::Local;
+using v8::MaybeLocal;
+using v8::Object;
+using v8::SnapshotCreator;
+using v8::String;
+using v8::Undefined;
+using v8::Value;
+
+Realm::Realm(IsolateData* isolate_data,
+             Environment* env,
+             v8::Local<v8::Context> context,
+             const RealmSerializeInfo* realm_info)
+    : isolate_data_(isolate_data), env_(env), isolate_(context->GetIsolate()) {
+  context_.Reset(isolate_, context);
+
+  if (realm_info != nullptr) {
+    DeserializeProperties(realm_info);
+    has_run_bootstrapping_code_ = true;
+  } else {
+    CreateProperties();
+  }
+}
+
+void Realm::MemoryInfo(MemoryTracker* tracker) const {
+#define V(PropertyName, TypeName)                                              \
+  tracker->TrackField(#PropertyName, PropertyName());
+  PER_REALM_STRONG_PERSISTENT_VALUES(V)
+#undef V
```

Comment:
You could track the isolate data and env here too (any maybe the context too, if TrackField() supports that)

Reply:
Yeah, updated with tracking isolate data and env. A v8::Context handle could not be tracked as it is not a subclass of v8::Value, so I'm leaving it as is.